streams/shared/src/main/scala/zio/stream/ZTransducer.scala

```diff
@@ -609,40 +610,50 @@ object ZTransducer {
   /**
    * Splits strings on a delimiter.
    */
-  def splitOn(delimiter: String): ZTransducer[Any, Nothing, String, String] =
+  def splitOn(delimiter: String): ZTransducer[Any, Nothing, String, String] = {
+    val chars = ZTransducer.fromFunction[String, Chunk[Char]](s => Chunk.fromArray(s.toArray)).mapChunks(_.flatten)
+    val split = splitOnChunk[Char, String](Chunk.fromArray(delimiter.toArray), _.mkString(""))
+    chars >>> split
+  }
+
+  /**
+   * Splits elements on a delimiter and transforms the splits into desired output.
+   */
+  def splitOnChunk[A, O: ClassTag](delimiter: Chunk[A], fn: Chunk[A] => O): ZTransducer[Any, Nothing, A, O] =
```

Comment:
Not sure we need `fn` here? That's easy enough for the user to apply with `ZTransducer#map`, no?

Reply:
Type signature would then be `ZTransducer[Any, Nothing, A, Chunk[A]]` that results in a `ZStream[R, E, Chunk[Chunk[A]]]`
Flattening the `Chunk[Chunk[A]]` would just reverse the split. So mapping the nested `Chunk[A]` is the only useful continuation I can think of.

Is there another useful thing to do with `Chunk[Chunk[A]]` than mapping it to some `Chunk[O]`?

Reply:
The resulting stream would be `ZStream[R, E, Chunk[A]]`, not `Chunk[Chunk[A]]` - applying `flattenChunks` would not reverse the split (because the delimiter is discarded) but rather provide a stream of all the individual elements.

Does that make sense?

Reply:
you are right. I was confused when working on a ZStream[R, E, Chunk[A]], but need to call `flattenChunks` in this case before splitting anyways.

changes applied already